crates/re_viewer_context/src/space_view/space_view_class.rs

```diff
@@ -0,0 +1,121 @@
+use std::any::Any;
+
+use crate::{SceneQuery, ViewerContext};
+use re_log_types::ComponentName;
+
+/// First element is the primary component, all others are optional.
+///
+/// TODO(andreas/clement): More formal definition of an archetype.
```

Comment:
Still wip of course, but you can get a sneak peek of the current state @ https://github.com/rerun-io/archetypes-codegen-poc/blob/master/out/rust/src/archetypes/point2d.rs